### PR TITLE
Milestone 3 : Game logic

### PR DESCRIPTION
**Pull request by tanzila-abedin**
1.    Created a branch name “game_logic”.
2.    Moved “puts”/“print” and “get.chomp” to main.rb file.
3.    Added a board class.
4.    Made use of “atrr_acessors”.

### CODE REVIEW

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :stop_sign: :red_circle: 
-------------
Hello, @tanzila-abedin @Stephanie041996  you are doing an amazing job on this milestone. :clap: 
However, there are a few changes to make to complete this milestone. :hammer_and_wrench: see them below :eyes: :point_down: 
------------
- [x] Please remove the file `.gitkeep` you are no longer need it. 
- [x] When I try to input any value on the board to select a position I got an error. 

Screenshot from 2021-04-06 17-28-47
---------------------
Please make the required changes and send it again for another review. :muscle: 
HAPPY CODING!!! :smile:

File: lib/board.rb
Comment (Berabjesus) [inline lib/board.rb]: - [x] [OPTIONAL] You dont need `attr_accessors` for `win_sequence` . You are not 'getting' or 'setting' it outside of the class.

File: bin/main
Comment (Berabjesus) [inline bin/main]: - [x] [Optional] You have a game logic in your main.rb file. 
All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

The `main` files should be responsible for displaying and getting inputs. In this case, the array that keeps hold of the moves should be in the `lib` directory so that the decision comes from the logic file.

**Review by Berabjesus — APPROVED** [review]
# **Status: PROJECT APPROVED** 
#### Excellent job on this project @tanzila-abedin  and @Stephanie041996. :fireworks: Keep it up. 

All the required tasks by the previous reviewer have been fulfilled in this project.

**However** there are a few bugs that went unnoticed. Please have a look at the following **OPTIONAL** comments and changes you should make. (I highly! recommend you implement them before merging).

- [x] [OPTIONAL] On your readme ->  getting started section -> line number 4, 5, 6 its specifying how one can make a contribution not how one can clone the repo and run the program. Should include something like `git checkout game_logic` and to run `ruby bin/main`
- [x] [OPTIONAL] unnessasary user of `attr_accessors` (check comment on the code below)
- [x] [OPTIONAL] The game has a bug when a user enters an already occupied spot. 
#### First user goes first (Correct) 👇 
image
#### second user goes next and hits the already ocupied key (1) gets this message 👇 
image

#### second user learns from mistake and hits the number 2 👇 
image

#### The board now changes the first `x` to and `o`. I belive this is a game logic issue and although its optional, I think you should fix it.

- [x] [Optional] You have a game logic in your main.rb file.  All game-logic related resources are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

- [x] [OPTIONAL]The game has a bug after a player enters some invalid input. The board will add a character to the ninth block.
image


- [x] [OPTIONAL] Keep all game logic-related code in the `lib` directory. (comment below)



I wish you all the very best in your next tasks:1st_place_medal:
#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** **slack(Bereket Beshane)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Pull request merged**
